pkg/flare/remote_config.go

```diff
@@ -0,0 +1,66 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2022-present Datadog, Inc.
+
+package flare
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/util"
+	"go.etcd.io/bbolt"
+)
+
+func stripRCTargets(raw []byte) []byte {
+	hash := sha256.Sum256(raw)
+	// Convert it to readable hex
+	s := hex.EncodeToString(hash[:])
+
+	return []byte(s)
+}
+
+func zipRemoteConfigDB(tempDir, hostname string) error {
+	dstPath := filepath.Join(tempDir, hostname, "remote-config.db")
+	srcPath := filepath.Join(config.Datadog.GetString("run_path"), "remote-config.db")
+
+	err := util.CopyFileAll(srcPath, dstPath)
+	if err != nil {
+		// Prevent from having a clear db here
+		os.Remove(dstPath)
+		return err
+	}
+
```

Comment:
can we add here a `defer os.Remove`? it will simplify the rest of the code

Reply:
We don't always want to delete the file at the end of the function, only when something goes wrong and we couldn't process the flare db